src/Core/src/ImageSources/StreamImageSourceService/StreamImageSourceService.Android.cs

```diff
@@ -24,15 +24,41 @@ public partial class StreamImageSourceService
 				{
 					stream = await streamImageSource.GetStreamAsync(cancellationToken);
 
-					var callback = new ImageLoaderCallback();
-
-					PlatformInterop.LoadImageFromStream(imageView, stream, callback);
-
-					var result = await callback.Result;
+					if (stream.CanSeek)
+					{
+						stream.Position = 0;
+					}
+					// Convert stream to byte array to avoid AndroidMarshalMethod InputStreamAdapter issues in release builds
+					byte[] buffer;
+					int length;
+					using (var memoryStream = new MemoryStream())
+					{
+						await stream.CopyToAsync(memoryStream, cancellationToken);
```

Comment:
This creates an extra copy of the entire image in memory. So, I'd expect it's going to 2x memory usage for every image loaded by a `Stream`.

Can we fix the underlying issue instead of this workaround?

If there is an issue still with `$(AndroidEnableMarshalMethods)=true`, we should either fix it turn the feature off by default in .NET 10.

Reply:
From a quick look, I doubt the run-time side of marshal methods is at fault here. I suspect it might be its build aspect (or it is a red herring). I'll investigate.